SV pipeline tries to create illegal interval

running wgs1 with current code in master, we get the following stack trace:

Caused by: java.lang.IllegalArgumentException: Invalid interval. Contig:chrUn_JTFH01001938v1_decoy start:0 end:0
	at org.broadinstitute.hellbender.utils.Utils.validateArg(Utils.java:686)
	at org.broadinstitute.hellbender.utils.SimpleInterval.validatePositions(SimpleInterval.java:60)
	at org.broadinstitute.hellbender.utils.SimpleInterval.<init>(SimpleInterval.java:36)
	at org.broadinstitute.hellbender.tools.spark.sv.discovery.NovelAdjacencyReferenceLocations$BreakpointsInference.getLeftJustifiedBreakpoints(NovelAdjacencyReferenceLocations.java:78)
	at org.broadinstitute.hellbender.tools.spark.sv.discovery.NovelAdjacencyReferenceLocations.leftJustifyBreakpoints(NovelAdjacencyReferenceLocations.java:293)
	at org.broadinstitute.hellbender.tools.spark.sv.discovery.NovelAdjacencyReferenceLocations.<init>(NovelAdjacencyReferenceLocations.java:42)
	at org.broadinstitute.hellbender.tools.spark.sv.discovery.DiscoverVariantsFromContigAlignmentsSAMSpark.lambda$discoverNovelAdjacencyFromChimericAlignments$7(DiscoverVariantsFromContigAlignmentsSAMSpark.java:409)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1351)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:294)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:161)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:300)
	at java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:42)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:438)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:147)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:79)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:47)
	at org.apache.spark.scheduler.Task.run(Task.scala:86)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:274)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)

--- a/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/BreakpointComplications.java
+++ b/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/BreakpointComplications.java
@@ -176,12 +176,12 @@ public final class BreakpointComplications {
     BreakpointComplications(final ChimericAlignment chimericAlignment, final byte[] contigSeq) {
 
         // if-else testing order matters here:
-        // if the two segments in the input chimeric alignment suggests simple translocation {@see ChimericAlignment#isNotSimpleTranslocation()}, then
+        // if the two segments in the input chimeric alignment suggests simple translocation {@see ChimericAlignment#isLikelySimpleTranslocation()}, then
         //  we need other types of evidence to fully resolve type of event, until then we can only say a novel adjacency (BND) is detected;
         // if the two segments in the input chimeric alignment map to the same chromosome, then
         //  the segment with lower coordinate on the locally-assembled contig could map to a higher reference coordinate region
         //  under two basic types of SV's: inversion (strand switch necessary) and "translocation" (no strand switch necessary)
-        final boolean suggestsSimpleTranslocation = !chimericAlignment.isNotSimpleTranslocation();
+        final boolean suggestsSimpleTranslocation = chimericAlignment.isLikelySimpleTranslocation();
         if (suggestsSimpleTranslocation) {
             initForSuspectedTranslocation(chimericAlignment, contigSeq);
         } else if (chimericAlignment.strandSwitch != StrandSwitch.NO_SWITCH) { // TODO: 9/9/17 the case involves an inversion, could be retired once same chr strand-switch BND calls are evaluated.

--- a/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/ChimericAlignment.java
+++ b/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/ChimericAlignment.java
@@ -175,7 +175,7 @@ public class ChimericAlignment {
             final ChimericAlignment chimericAlignment = new ChimericAlignment(current, next, insertionMappings,
                     alignedContig.contigName, referenceDictionary);
             // the following check/filter is due to the fact that simple translocations are to be handled in a different code path
-            if (chimericAlignment.isNotSimpleTranslocation())
+            if (chimericAlignment.isNeitherSimpleTranslocationNorIncompletePicture())
                 results.add(chimericAlignment);
 
             current = next;
@@ -276,7 +276,7 @@ public class ChimericAlignment {
     }
 
     /**
-     * See {@link #isNotSimpleTranslocation()} for logic.
+     * See {@link #isLikelySimpleTranslocation()} for logic.
      */
     @VisibleForTesting
     public static boolean isLikelySimpleTranslocation(final AlignmentInterval regionWithLowerCoordOnContig,
@@ -319,8 +319,66 @@ public class ChimericAlignment {
      * which could be a translocation or inversion breakpoint.
      * But to fully resolve this case, we need other types of evidence, hence should not be the task of this function.
      */
-    public boolean isNotSimpleTranslocation() {
-        return !isLikelySimpleTranslocation(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, strandSwitch);
+    public boolean isLikelySimpleTranslocation() {
+        return isLikelySimpleTranslocation(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, strandSwitch);
+    }
+
+    /**
+     * This predicate tests if an assembly contig has the full event (i.e. alt haplotype) assembled
+     * Of course, the grand problem of SV is always not getting the big-enough picture
+     * but here we have a more workable definition of what is definitely not big-enough:
+     *
+     * If the assembly contig, with its two (picked) alignments, shows any of the following signature,
+     * it is definitely not giving the whole picture of the alt haplotype,
+     * hence without other types of evidence (or linking breakpoints, which itself needs other evidence anyway),
+     * human-friendly interpretation for them is unreliable.
+     * <ul>
+     *     <li>
+     *         first and second alignment contain each other in terms of their reference span,
+     *         regardless if strand switch is involved;
+     *     </li>
+     *     <li>
+     *         todo: this obsoletes the actual use of {@link #isLikelyInvertedDuplication()} and related inverted duplication call code we have now, but those could be used to figure out how to annotate which known ref regions are invert duplicated
+     *         first and second alignment involve strand switch, but their reference span overlap;
+     *     </li>
+     *     <li>
+     *         first and second alignment have reference order switch but their reference span overlaps;
+     *     </li>
+     * </ul>
+     */
+    public static boolean hasIncompletePictureFromTwoAlignments(final AlignmentInterval one, final AlignmentInterval two) {
+        final SimpleInterval referenceSpanOne = one.referenceSpan;
+        final SimpleInterval referenceSpanTwo = two.referenceSpan;
+
+        // inter contig mapping will not be treated as incomplete picture for 2-alignment reads
+        if ( ! referenceSpanOne.getContig().equals(referenceSpanTwo.getContig()) )
+            return false;
+
+        // ref span containment
+        if (referenceSpanOne.contains(referenceSpanTwo) || referenceSpanTwo.contains(referenceSpanOne))
+            return true;
+
+        // overlapping strand-switch
+        // TODO: 10/29/17 this obsoletes the inverted duplication call code we have now,
+        //      but those could be used to figure out how to annotate which known ref regions are invert duplicated
+        if (one.forwardStrand != two.forwardStrand &&
+                referenceSpanOne.overlaps(referenceSpanTwo))
+            return true;
+
+        // no strand switch but overlapping reference order switch
+        if (one.forwardStrand) {
+            return referenceSpanOne.getStart() > referenceSpanTwo.getStart() &&
+                    referenceSpanOne.getStart() <= referenceSpanTwo.getEnd();
+        } else {
+            return referenceSpanTwo.getStart() > referenceSpanOne.getStart() &&
+                    referenceSpanTwo.getStart() <= referenceSpanOne.getEnd();
+        }
+    }
+
+    public boolean isNeitherSimpleTranslocationNorIncompletePicture() {
+        return !isLikelySimpleTranslocation()
+                &&
+                !hasIncompletePictureFromTwoAlignments(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig);
     }
 
     /**

--- a/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigWithFineTunedAlignments.java
+++ b/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigWithFineTunedAlignments.java
@@ -7,6 +7,7 @@ import com.esotericsoftware.kryo.io.Input;
 import com.esotericsoftware.kryo.io.Output;
 import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignedContig;
 import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignmentInterval;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.ChimericAlignment;
 import org.broadinstitute.hellbender.utils.SimpleInterval;
 import org.broadinstitute.hellbender.utils.Utils;
 
@@ -63,48 +64,11 @@ final class AssemblyContigWithFineTunedAlignments {
     //==================================================================================================================
 
     boolean hasIncompletePictureFromTwoAlignments() {
-        return hasIncompletePictureDueToRefSpanContainment()
-                ||
-                (firstAndLastAlignmentMappedToSameChr() && hasIncompletePictureDueToOverlappingRefOrderSwitch());
-    }
-
-    boolean hasIncompletePictureDueToRefSpanContainment() {
-
-        final AlignmentInterval one = contig.alignmentIntervals.get(0);
-        final AlignmentInterval two = contig.alignmentIntervals.get(1);
-        return one.referenceSpan.contains(two.referenceSpan)
-                ||
-                two.referenceSpan.contains(one.referenceSpan);
-    }
-
-    boolean hasIncompletePictureDueToOverlappingRefOrderSwitch() {
-
-        final AlignmentInterval one = contig.alignmentIntervals.get(0);
-        final AlignmentInterval two = contig.alignmentIntervals.get(1);
-        final SimpleInterval referenceSpanOne = one.referenceSpan;
-        final SimpleInterval referenceSpanTwo = two.referenceSpan;
-
-        if (referenceSpanOne.contains(referenceSpanTwo) || referenceSpanTwo.contains(referenceSpanOne))
-            return true;
-
-        // TODO: 10/29/17 this obsoletes the inverted duplication call code we have now, but those could be used to figure out how to annotate which known ref regions are invert duplicated
-        if (one.forwardStrand != two.forwardStrand) {
-            return referenceSpanOne.overlaps(referenceSpanTwo);
-        } else {
-            if (one.forwardStrand) {
-                return referenceSpanOne.getStart() > referenceSpanTwo.getStart() &&
-                        referenceSpanOne.getStart() <= referenceSpanTwo.getEnd();
-            } else {
-                return referenceSpanTwo.getStart() > referenceSpanOne.getStart() &&
-                        referenceSpanTwo.getStart() <= referenceSpanOne.getEnd();
-            }
-        }
+        return ChimericAlignment.hasIncompletePictureFromTwoAlignments(contig.alignmentIntervals.get(0),
+                                                                        contig.alignmentIntervals.get(1));
     }
 
     boolean firstAndLastAlignmentMappedToSameChr() {
-        Utils.validateArg(!hasIncompletePictureDueToRefSpanContainment(),
-                "assumption that input contig has alignments whose ref span is not completely covered by the other's is violated \n" +
-                        contig.toString());
 
         final String firstMappedChr = this.contig.alignmentIntervals.get(0).referenceSpan.getContig();
         final String lastMappedChr  = this.contig.alignmentIntervals.get(this.contig.alignmentIntervals.size() - 1).referenceSpan.getContig();

--- a/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InsDelVariantDetector.java
+++ b/src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InsDelVariantDetector.java
@@ -20,8 +20,6 @@ import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 
-import static org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection.CHIMERIC_ALIGNMENTS_HIGHMQ_THRESHOLD;
-
 
 final class InsDelVariantDetector implements VariantDetectorFromLocalAssemblyContigAlignments {
 
@@ -78,7 +76,8 @@ final class InsDelVariantDetector implements VariantDetectorFromLocalAssemblyCon
             }
 
             final ChimericAlignment ca = new ChimericAlignment(current, next, insertionMappings, contig.contigName, referenceDictionary);
-            if (!ca.isNotSimpleTranslocation())
+            final boolean validStateForThisPath = ca.isNeitherSimpleTranslocationNorIncompletePicture();
+            if ( ! validStateForThisPath )
                 throw new GATKException.ShouldNeverReachHereException("Mapped assembled contigs are sent down the wrong path: " +
                         "contig suggesting \"translocation\" is sent down the insert/deletion path.\n" + contig.toString());
             results.add(ca);

--- a/src/test/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/ChimericAlignmentUnitTest.java
+++ b/src/test/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/ChimericAlignmentUnitTest.java
@@ -55,101 +55,101 @@ public class ChimericAlignmentUnitTest extends GATKBaseTest {
 
         // simple inversion
         Tuple4<AlignmentInterval, AlignmentInterval, NovelAdjacencyReferenceLocations, String> testData = forSimpleInversionFromLongCtg1WithStrangeLeftBreakpoint;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.REVERSE_TO_FORWARD, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.REVERSE_TO_FORWARD, true, false, false, false, false});
 
         testData = forSimpleInversionWithHom_leftPlus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.FORWARD_TO_REVERSE, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.FORWARD_TO_REVERSE, true, false, false, false, false});
 
         testData = forSimpleInversionWithHom_leftMinus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.FORWARD_TO_REVERSE, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.FORWARD_TO_REVERSE, false, false, false, false, false});
 
         testData = forSimpleInversionWithHom_rightPlus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.REVERSE_TO_FORWARD, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.REVERSE_TO_FORWARD, true, false, false, false, false});
 
         testData = forSimpleInversionWithHom_rightMinus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.REVERSE_TO_FORWARD, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.REVERSE_TO_FORWARD, false, false, false, false, false});
 
         // simple deletion
         testData = SVDiscoveryTestDataProvider.forSimpleDeletion_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forSimpleDeletion_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
         // simple insertion
         testData = SVDiscoveryTestDataProvider.forSimpleInsertion_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forSimpleInsertion_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
         // long range substitution
         testData = SVDiscoveryTestDataProvider.forLongRangeSubstitution_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forLongRangeSubstitution_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
         // simple deletion with homology
         testData = SVDiscoveryTestDataProvider.forDeletionWithHomology_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forDeletionWithHomology_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
         // tandem duplication simple contraction
         testData = SVDiscoveryTestDataProvider.forSimpleTanDupContraction_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forSimpleTanDupContraction_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
         // tandem duplication simple expansion
         testData = SVDiscoveryTestDataProvider.forSimpleTanDupExpansion_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forSimpleTanDupExpansion_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
         // tandem duplication simple expansion with novel insertion
         testData = SVDiscoveryTestDataProvider.forSimpleTanDupExpansionWithNovelIns_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forSimpleTanDupExpansionWithNovelIns_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
 
 
         // first test (the original observed event, but assigned to a different chromosome): expansion from 1 unit to 2 units with pseudo-homology
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_1to2_pseudoHom_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_1to2_pseudoHom_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
 
         // second test: contraction from 2 units to 1 unit with pseudo-homology
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_2to1_pseudoHom_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_2to1_pseudoHom_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
 
         // third test: contraction from 3 units to 2 units without pseudo-homology
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_3to2_noPseudoHom_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_3to2_noPseudoHom_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
 
         // fourth test: expansion from 2 units to 3 units without pseudo-homology
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_2to3_noPseudoHom_plus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, true, false, false, false, false});
 
         testData = SVDiscoveryTestDataProvider.forComplexTanDup_2to3_noPseudoHom_minus;
-        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false});
+        data.add(new Object[]{testData._1(), testData._2(), StrandSwitch.NO_SWITCH, false, false, false, false, false});
 
         ////////// ABOVE ARE FOR SIMPLE VARIANTS: INS/DEL, DUP EXPANSION, DUP CONTRACTION, INVERSION, BELOW ARE FOR TRANSLOCATION SUSPECTS AND INV DUP
 
@@ -175,32 +175,36 @@ public class ChimericAlignmentUnitTest extends GATKBaseTest {
         AlignmentInterval intervalOne = new AlignmentInterval(one);
         AlignmentInterval intervalTwo = new AlignmentInterval(two);
 
-        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.REVERSE_TO_FORWARD, false, false, true, true});
+        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.REVERSE_TO_FORWARD, false, false, true, true, true});
 
         intervalOne = new AlignmentInterval(new SimpleInterval("chr20", 48513458, 48513545), 1, 88, TextCigarCodec.decode("88M227H"), true, 39, 1, 83, AlnModType.NONE);
         intervalTwo = new AlignmentInterval(new SimpleInterval("chr20", 48513297, 48513579), 84, 365, TextCigarCodec.decode("83S282M"), false, 60, 0, 282, AlnModType.NONE);
-        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.FORWARD_TO_REVERSE, true, false, true, true});
+        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.FORWARD_TO_REVERSE, true, false, true, true, true});
 
 
         // same-chr translocation suspect, forward and reverse representation
         intervalOne = new AlignmentInterval(new SimpleInterval("chr20", 61015129, 61015272), 1, 144, TextCigarCodec.decode("144M148H"), true, 60, 1, 139, AlnModType.NONE);
         intervalTwo = new AlignmentInterval(new SimpleInterval("chr20", 60992732, 60992880), 144, 292, TextCigarCodec.decode("143S149M"), true, 60, 0, 149, AlnModType.NONE);
-        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.NO_SWITCH, true, true, false, true});
+        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.NO_SWITCH, true, true, false, false, true});
 
         intervalOne = new AlignmentInterval(new SimpleInterval("chr20", 28861368, 28861775), 1, 409, TextCigarCodec.decode("387M1I21M623H"), false, 60, 22, 286, AlnModType.NONE);
         intervalTwo = new AlignmentInterval(new SimpleInterval("chr20", 28896473, 28897229), 276, 1032, TextCigarCodec.decode("275S757M"), false, 60, 1, 752, AlnModType.NONE);
-        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.NO_SWITCH, false, true, false, true});
+        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.NO_SWITCH, false, true, false, false, true});
 
         // diff-chr translocation suspect without SS
         intervalOne = new AlignmentInterval(new SimpleInterval("chr21", 24923683, 24923715), 1, 33, TextCigarCodec.decode("33M130H"), true, 60, 0, 33, AlnModType.NONE);
         intervalTwo = new AlignmentInterval(new SimpleInterval("chr20", 11590055, 11590197), 21, 163, TextCigarCodec.decode("20S143M"), true, 60, 3, 128, AlnModType.NONE);
-        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.NO_SWITCH, true, true, false, true});
+        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.NO_SWITCH, true, true, false, false, true});
 
         // diff-chr translocation suspect with SS
         intervalOne = new AlignmentInterval(new SimpleInterval("chr21", 5374092, 5374748), 1, 656, TextCigarCodec.decode("656M322S"), true, 60, 14, 586, AlnModType.NONE);
         intervalTwo = new AlignmentInterval(new SimpleInterval("chr20", 28764673, 28765145), 506, 978, TextCigarCodec.decode("473M505H"), false, 60, 16, 393, AlnModType.NONE);
-        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.FORWARD_TO_REVERSE, false, true, false, true});
+        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.FORWARD_TO_REVERSE, false, true, false, false, true});
 
+        // same-chr reference order switch, but overlaps (hence incomplete picture)
+        intervalOne = new AlignmentInterval(new SimpleInterval("20", 283, 651), 383, 751, TextCigarCodec.decode("382H369M274H"), true, 60, 23, 254, AlnModType.NONE);
+        intervalTwo = new AlignmentInterval(new SimpleInterval("20", 1, 413), 613, 1025, TextCigarCodec.decode("612H413M"), true, 60, 0, 413, AlnModType.NONE);
+        data.add(new Object[]{intervalOne, intervalTwo, StrandSwitch.NO_SWITCH, true, false, false, true, true});
 
         return data.toArray(new Object[data.size()][]);
     }
@@ -211,6 +215,7 @@ public class ChimericAlignmentUnitTest extends GATKBaseTest {
                      final boolean expectedIsForwardStrandRepresentation,
                      final boolean expectedIsLikelySimpleTranslocation,
                      final boolean expectedIsLikelyInvDup,
+                     final boolean expectedIsIncompletePicture,
                      final boolean shouldUseRefVersion_b38) {
 
         final SAMSequenceDictionary refDict = shouldUseRefVersion_b38 ? SVDiscoveryTestDataProvider.b38_seqDict : SVDiscoveryTestDataProvider.seqDict;
@@ -219,6 +224,7 @@ public class ChimericAlignmentUnitTest extends GATKBaseTest {
         Assert.assertEquals(ChimericAlignment.isForwardStrandRepresentation(region1, region2, expectedStrandSwitch, refDict), expectedIsForwardStrandRepresentation);
         Assert.assertEquals(ChimericAlignment.isLikelySimpleTranslocation(region1, region2, expectedStrandSwitch), expectedIsLikelySimpleTranslocation);
         Assert.assertEquals(ChimericAlignment.isLikelyInvertedDuplication(region1, region2), expectedIsLikelyInvDup);
+        Assert.assertEquals(ChimericAlignment.hasIncompletePictureFromTwoAlignments(region1, region2), expectedIsIncompletePicture);
 
         final ChimericAlignment chimericAlignment = new ChimericAlignment(region1, region2, Collections.emptyList(), "dummyName", refDict);
         final ByteArrayOutputStream bos = new ByteArrayOutputStream();

